srml/support/src/traits.rs

```diff
@@ -51,23 +53,57 @@ impl<Balance> OnDilution<Balance> for () {
 	fn on_dilution(_minted: Balance, _portion: Balance) {}
 }
 
-/// Determinator for whether a given account is able to transfer balance.
-pub trait EnsureAccountLiquid<AccountId> {
-	/// Returns `Ok` iff the account is able to transfer funds normally. `Err(...)`
-	/// with the reason why not otherwise.
+/// Determinator for whether a given account is able to use its **free** balance.
+/// 
+/// By convention, `ensure_account_liquid` overrules `ensure_account_can_withdraw`. If a
+/// caller gets `Ok` from the former, then they do not need to call the latter.
+/// 
+/// This implies that if you define the latter away from its default of replicating the
+/// former, then ensure you also redefine the former to return an `Err` in corresponding
+/// situations, otherwise you'll end up giving inconsistent information.
+pub trait EnsureAccountLiquid<AccountId, Balance> {
+	/// Ensures that the account is completely unencumbered. If this is `Ok` then there's no need to
+	/// check any other items. If it's an `Err`, then you must use one pair of the other items.
```

Comment:
what is the reason to have both `ensure_account_liquid` and `ensure_account_can_withdraw`?
`ensure_account_liquid` is not used anywhere in substrate anymore.
having both make implementation harder. more confusion on usage. I cannot think any case where I should call `ensure_account_liquid` over `ensure_account_can_withdraw`

Reply:
`ensure_account_liquid` is still used (afaik) in srml contract module.

Reply:
which should be changed to call `ensure_account_can_withdraw` with `Transfer` reason?

Reply:
No, because the contracts module mutates the balance locally, yet `EnsureAccountLiquid` relies on `free_balance` being correct. It could perhaps be done by passing an additional `balance_override: Option<Balance>` parameter, but since it's all likely to be refactored anyway in #1896 , then it seems redundant to introduce here.